contracts/switchboard/default-switchboards/FastSwitchboard.sol

```diff
@@ -29,59 +26,62 @@ contract FastSwitchboard is SwitchboardBase {
     event SocketSet(address newSocket);
     // Event emitted when a packet is attested
     event PacketAttested(bytes32 packetId, address attester);
-    // Event emitted when the attest gas limit is set
-    event AttestGasLimitSet(uint256 dstChainSlug, uint256 attestGasLimit);
 
     // Error emitted when a watcher is found
     error WatcherFound();
     // Error emitted when a watcher is not found
     error WatcherNotFound();
     // Error emitted when a packet is already attested
     error AlreadyAttested();
+    // Error emitted when role is invalid
+    error InvalidRole();
 
     /**
      * @dev Constructor function for the FastSwitchboard contract
      * @param owner_ Address of the owner of the contract
      * @param socket_ Address of the socket contract
-     * @param gasPriceOracle_ Address of the gas price oracle contract
      * @param chainSlug_ Chain slug of the chain where the contract is deployed
      * @param timeoutInSeconds_ Timeout in seconds for the packets
+     * @param signatureVerifier_ The address of the signature verifier contract
      */
     constructor(
         address owner_,
         address socket_,
-        address gasPriceOracle_,
-        uint256 chainSlug_,
-        uint256 timeoutInSeconds_
+        uint32 chainSlug_,
+        uint256 timeoutInSeconds_,
+        ISignatureVerifier signatureVerifier_
     )
         AccessControlExtended(owner_)
-        SwitchboardBase(gasPriceOracle_, socket_, chainSlug_, timeoutInSeconds_)
+        SwitchboardBase(
+            socket_,
+            chainSlug_,
+            timeoutInSeconds_,
+            signatureVerifier_
+        )
     {}
 
     /**
      * @dev Function to attest a packet
      * @param packetId_ Packet ID
-     * @param srcChainSlug_ Chain slug of the source chain
      * @param signature_ Signature of the packet
      */
-    function attest(
-        bytes32 packetId_,
-        uint256 srcChainSlug_,
-        bytes calldata signature_
-    ) external {
+    function attest(bytes32 packetId_, bytes calldata signature_) external {
+        uint32 srcChainSlug = uint32(uint256(packetId_) >> 224);
```

Comment:
why use `SignatureVerifierLib.recoverSignerFromDigest()` below
and not `signatureVerifier__.recoverSignerFromDigest()`
(also in other places)